src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -150,6 +150,7 @@ private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, I
                 (argument) =>
                 {
                     string name = !argumentNames.IsDefaultOrEmpty ? argumentNames[argumentIndex] : null;
+                    Symbols.ParameterSymbol parameter = parameterIndex < parameters.Length ? parameters[parameterIndex] : null;
```

Comment:
I think we can't return `boundArguments[argumentIndex]` since it is not an `IArgument`. We sure can bail early by creating and returning an `Argument` if HasErrors is true. Either way is OK to me.